discovery/cmd/openstack-discoverer/main.go

```diff
@@ -118,18 +121,33 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed to create OpenStack client: %v", err)
 	}
-	osClient.HTTPClient.Timeout = httpClientTimeout
+
+	dClient := http.Client{
+		Transport: &openstack.LogRoundTripper{
+			RoundTripper: http.DefaultTransport,
+			Log:          log,
+			ClusterName:  clusterName,
+			ClusterType:  clusterType,
+			Metrics:      &discovererMetrics,
+		},
+		Timeout: httpClientTimeout,
+	}
+
 	if openstackCertificateAuthorityFile != "" {
-		osClient.HTTPClient.Transport = httpTransportWithCA(log, openstackCertificateAuthorityFile)
+		dClient.Transport = httpTransportWithCA(log, openstackCertificateAuthorityFile)
```

Comment:
We will overwrite the `openstack.LogRoundTripper` that we are configuring in L126. 

I think something like this would work?
```
transport := &openstack.LogRoundTripper{
			RoundTripper: http.DefaultTransport,
			Log:          log,
			ClusterName:  clusterName,
			ClusterType:  clusterType,
			Metrics:      &discovererMetrics,
}
if openstackCertificateAuthorityFile != "" {
  transport.RoundTripper = httpTransportWithCA(log, openstackCertificateAuthorityFile)
}

osClient.HTTPClient = http.Client{
  Transport: transport,
  Timeout: httpClientTimeout,
}
```